### PR TITLE
Add user guide

### PR DESCRIPTION
**Pull request by HuangJiameng**

### CODE REVIEW

**Comment by HuangJiameng** [comment]
Add a user guide to help the new user get involved and contribute.
HD video is ready now.

File: doc/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# User Guide
+## Discussions：
+
+<div align=center><href="https://sm.ms/image/DVrYCuwFlM4Pqni" target="_blank"><img src="https://s2.loli.net/2022/08/16/DVrYCuwFlM4Pqni.png" width="60%"></div>
+
+Welcome everyone to participate in the discussion about DP-GEN in the discussion module. You can ask for help, share an idea or anything to discuss here. The website is https://github.com/deepmodeling/dpgen/discussions
```
Comment (njzjz) [inline doc/user-guide.md:6]: image

Please make all links "blue".

File: doc/how-to-contribute.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+3. Import https://github.com/<your-github-username>/dpgen.git 
+Remember to set Project homepage as https://github.com/<your-github-username>/dpgen , and set the Default branch as the one you do modification in Advanced Settings.
+
+<div align=center><href="https://sm.ms/image/4cVRb7ytT1h5l9n" target="_blank"><img src="https://s2.loli.net/2022/08/16/4cVRb7ytT1h5l9n.png" width="60%"></div>
+
+(“doc_overview” is just an example, please change it into your branch's name.)
+
+4. Build Version 
+
+<div align=center><href="https://sm.ms/image/CsJ7S5xeTYh9EWp" target="_blank"><img src="https://s2.loli.net/2022/08/16/CsJ7S5xeTYh9EWp.png" width="60%"></div>
```
Comment (njzjz) [inline doc/how-to-contribute.md:114]: It's not necessary as we have PR build:
image

File: doc/how-to-contribute.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+and [Developer guide](https://github.com/deepmodeling/deepmd-kit/edit/devel/doc/development/index.md).
+
+Welcome to the repository of DP-GEN https://github.com/deepmodeling/dpgen
+
+  If you have no idea how to fix your problem or where to find the relative code, see OVERVIEW/Overview of the structure of the DP-GEN repository on this website.
+  
+### Use command line
+You can use git with the command line, or open the repository on Github Desktop. Here is a video as a demo of making changes to DP-GEN and publishing it with command line.
+
+https://www.youtube.com/watch?v=DPApc1_eNS8
```
Comment (njzjz) [inline doc/how-to-contribute.md:26]: You can use my script [bilitube](https://github.com/njzjz/bilitube) to automatically switch between bilibili and youtube.

File: doc/how-to-contribute.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+Firstly, fork in DP-GEN repository. Then you can clone the repository, build a new branch, make changes and then make a pull request.
+
+---
+
+## How to contribute to DP-GEN
+
+DP-GEN adopts the same convention as other softwares in DeepModeling Community. 
+
+You can first refer to DeePMD-kit's
+[Contributing guide](https://github.com/deepmodeling/deepmd-kit/edit/devel/CONTRIBUTING.md)
```
Comment (njzjz) [inline doc/how-to-contribute.md:16]: This link is to "edit"?

File: doc/how-to-contribute.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+and [Developer guide](https://github.com/deepmodeling/deepmd-kit/edit/devel/doc/development/index.md).
```
Comment (njzjz) [inline doc/how-to-contribute.md:17]: And also this link.

File: doc/how-to-contribute.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+----
+  
+## Tips
+  
+1. Please try to submit a PR after finishing all the changes
+  
+2. Please briefly describe what you do with `git commit -m "what you do"`! "No description provided." will make the maintainer feel confused.
+
+3. It is not recommended to make changes directly in the `devel` branch. It is recommended to pull a branch from devel: `git is git checkout -b <new-branch-name>`
```
Comment (njzjz) [inline doc/how-to-contribute.md:104]: Is `git is` a typo?

File: doc/how-to-contribute.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+<div align=center><href="https://sm.ms/image/qt1MWOfYbQzKGJC" target="_blank"><img src="https://s2.loli.net/2022/08/16/qt1MWOfYbQzKGJC.png" width="60%"></div>
+
+Firstly, fork in DP-GEN repository. Then you can clone the repository, build a new branch, make changes and then make a pull request.
+
+---
+
+## How to contribute to DP-GEN
+
+DP-GEN adopts the same convention as other softwares in DeepModeling Community.
```
Comment (njzjz) [inline doc/how-to-contribute.md:13]: softwares -> software

Comment (njzjz) [reply doc/how-to-contribute.md:13]: software is uncountable

File: doc/how-to-contribute.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+You can first refer to DeePMD-kit's
+[Contributing guide](https://github.com/deepmodeling/deepmd-kit/blob/master/CONTRIBUTING.md)
+and [Developer guide](https://docs.deepmodeling.com/projects/deepmd/en/master/development/coding-conventions.html).
+
+You can also read relative chapters on [Github Docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).
+
+Welcome to the repository of [DP-GEN](https://github.com/deepmodeling/dpgen) 
+
+If you have no idea how to fix your problem or where to find the relative code, see OVERVIEW/Overview of the structure of the DP-GEN repository on this website.
```
Comment (njzjz) [inline doc/how-to-contribute.md:23]: add a link?

File: doc/user-guide.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+## Example for parameters
+      
+If you have no idea how to prepare a `PARAM` for your task, you can find examples of PARAM for different tasks in [examples](https://github.com/deepmodeling/dpgen/tree/master/examples).
+      
+For example, if you want to set specific template for LAMPPS, you can find an example [here](https://github.com/deepmodeling/dpgen/blob/master/examples/run/dp-lammps-enhance_sampling/param.json)
+      
+If you want to learn more about Machine parameters, please check [docs for dpdispatcher](https://docs.deepmodeling.com/projects/dpdispatcher/en/latest/)
+      
+## [Pull requests - How to contribute](https://docs.deepmodeling.com/projects/dpgen/en/latest/how-to-contribute.html)
```
Comment (njzjz) [inline doc/user-guide.md:33]: You can use a relative URL... Is there a problem?

File: doc/user-guide.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      
+## Tutorials 
+
+Tutorials can be found [here](https://tutorials.deepmodeling.com/en/latest/Tutorials/DP-GEN/index.html).
+
+## Example for parameters
+      
+If you have no idea how to prepare a `PARAM` for your task, you can find examples of PARAM for different tasks in [examples](https://github.com/deepmodeling/dpgen/tree/master/examples).
+      
+For example, if you want to set specific template for LAMPPS, you can find an example [here](https://github.com/deepmodeling/dpgen/blob/master/examples/run/dp-lammps-enhance_sampling/param.json)
```
Comment (njzjz) [inline doc/user-guide.md:29]: LAMMPS

File: doc/user-guide.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+## Issue：
+
+<div align=center><href="https://sm.ms/image/Tu1dRX2L6zO5Yfq" target="_blank"><img src="https://s2.loli.net/2022/08/16/Tu1dRX2L6zO5Yfq.png" width="60%"></div>    
+      
+If you want to make a bug report or a request for new feature, you can make an issue in the issue module. 
+
+<div align=center><href="https://sm.ms/image/FajXcmwbBJo34y1" target="_blank"><img src="https://s2.loli.net/2022/08/16/FajXcmwbBJo34y1.png" width="60%"></div>  
+  
+Here are the types you can choose. A proper type can help developer figure out what you need. Also, you can assign yourself to solve the issue. Your contribution is welcomed!
```
Comment (njzjz) [inline doc/user-guide.md:17]: welcomed -> welcome

File: doc/user-guide.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+<div align=center><href="https://sm.ms/image/DVrYCuwFlM4Pqni" target="_blank"><img src="https://s2.loli.net/2022/08/16/DVrYCuwFlM4Pqni.png" width="60%"></div>
+
+Welcome everyone to participate in the discussion about DP-GEN in the [discussion](https://github.com/deepmodeling/dpgen/discussions) module. You can ask for help, share an idea or anything to discuss here.
+Note: before you raise a question, please check TUTORIAL/FAQs and search history discussions to find solutions. 
+
+## Issue：
+
+<div align=center><href="https://sm.ms/image/Tu1dRX2L6zO5Yfq" target="_blank"><img src="https://s2.loli.net/2022/08/16/Tu1dRX2L6zO5Yfq.png" width="60%"></div>    
+      
+If you want to make a bug report or a request for new feature, you can make an issue in the issue module.
```
Comment (njzjz) [inline doc/user-guide.md:13]: new feature -> new features / a new feature

File: doc/user-guide.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+## Example for parameters
+      
+If you have no idea how to prepare a `PARAM` for your task, you can find examples of PARAM for different tasks in [examples](https://github.com/deepmodeling/dpgen/tree/master/examples).
+      
+For example, if you want to set specific template for LAMMPS, you can find an example [here](https://github.com/deepmodeling/dpgen/blob/master/examples/run/dp-lammps-enhance_sampling/param.json)
+      
+If you want to learn more about Machine parameters, please check [docs for dpdispatcher](https://docs.deepmodeling.com/projects/dpdispatcher/en/latest/)
+      
+## [Pull requests - How to contribute](./how-to-contribute.html)
```
Comment (njzjz) [inline doc/user-guide.md:33]: emmmm... The link does not work. I will try to find a solution if you fail to resolve it.

Comment (HuangJiameng) [reply doc/user-guide.md]: I followed [answer-1444](https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file). The relative link works in `Preview` but doesn't work in webpage.

**Review by njzjz — APPROVED** [review]

File: doc/user-guide.md (hunk 13 above)
Comment (njzjz) [reply doc/user-guide.md]: I push f86d56bd56031dbbb32c21d3149ed5944a12fd53 to resolve the issue. You can take a look.

**Pull request merged**